components/backup-stream/src/endpoint.rs

```diff
@@ -319,19 +319,41 @@ where
         scheduler: Scheduler<Task>,
         revision: i64,
     ) -> Result<()> {
-        let mut watcher = meta_client.events_from(revision).await?;
+        let mut revision_new = revision;
         loop {
-            if let Some(event) = watcher.stream.next().await {
-                info!("backup stream watch event from etcd"; "event" => ?event);
-                match event {
-                    MetadataEvent::AddTask { task } => {
-                        scheduler.schedule(Task::WatchTask(TaskOp::AddTask(task)))?;
+            let watcher = meta_client.events_from(revision_new).await;
+            let mut watcher = match watcher {
+                Ok(w) => w,
+                Err(e) => {
+                    e.report("failed to start watch pause");
+                    tokio::time::sleep(Duration::from_secs(5)).await;
+                    continue;
+                }
+            };
+
+            loop {
+                if let Some(event) = watcher.stream.next().await {
```

Comment:
if `watcher.stream.next().await` returns `None` this will become a hot infinite loop?